NvimView/Sources/NvimView/NvimView+RemoteOptions.swift

```diff
@@ -0,0 +1,129 @@
+/**
+ * Renee Koecher -  @shirk
+ * See LICENSE
+ */
+
+import Cocoa
+import MessagePack
+import RxSwift
+
+extension NvimView {
+  enum RemoteOption {
+    // list of currently handled remote options
+    case guifont(fontSpec: String)
+    case guifontWide(fontSpec: String)
+
+    static func fromValuePair(_ option: (key: MessagePackValue, value: MessagePackValue)) -> RemoteOption? {
+      switch option.key.stringValue ?? "" {
+      case "guifont": return RemoteOption.guifont(fontSpec: option.value.stringValue ?? "")
+      case "guifontwide": return RemoteOption.guifontWide(fontSpec: option.value.stringValue ?? "")
+
+      default: return nil
+      }
+    }
+
+    // convenience methods
+    static func fromFont(_ font: NSFont, forWideFont isWide: Bool = false) -> RemoteOption? {
+      guard let fontSpec = FontUtils.vimFontSpec(forFont: font) else {
+        return nil
+      }
+
+      if isWide {
+        return RemoteOption.guifontWide(fontSpec: fontSpec)
+      }
+      return RemoteOption.guifont(fontSpec: fontSpec)
+    }
+  }
+
+  final func handleRemoteOptions(_ options: [MessagePackValue: MessagePackValue]) {
+    for kvPair in options {
+      guard let option = RemoteOption.fromValuePair(kvPair) else {
+        self.bridgeLogger.debug("Could not handle RemoteOption \(kvPair)")
+        continue
+      }
+
+      switch option {
+        // fixme: currently this treats gft and gfw the as the same
+        case .guifont(let fontSpec): handleGuifontSet(fontSpec)
+        case .guifontWide(let fontSpec): handleGuifontSet(fontSpec, forWideFont: true)
+      }
+    }
+  }
+
+  final func signalRemoteOptionChange(_ option: RemoteOption) {
+    var command: Completable? = nil
+
+    switch option {
+    case .guifont(let fontSpec):
+        command = self.api.setOption(name: "guifont", value: .string(fontSpec))
+
+    case .guifontWide(let fontSpec):
+        command = self.api.setOption(name: "guifontwide", value: .string(fontSpec))
+    }
+
+    command?.subscribe().disposed(by: self.disposeBag)
+  }
+
+  public final func signalError(code: Int, message: String) {
+      self.api.errWriteln(str: "E\(code): \(message)")
```

Comment:
I didn't even know that this function existed :) Good to know.

Reply:
Took me a bit of rooting through the Nvim API but yeah :D
Currently I'm sending the same error code for fonts that gVIM does, this could be changed though.